front-end/src/app/shared/components/transaction-type-base/triple-transaction-type-base.component.ts

```diff
@@ -0,0 +1,214 @@
+    ) {
+      this.childMemoCodeCheckboxLabel_2$ = this.memoCodeCheckboxLabel$;
+    } else {
+      this.childMemoCodeCheckboxLabel_2$ = this.getMemoCodeCheckboxLabel$(
+        this.childForm_2,
+        this.childTransactionType_2
+      );
+    }
+
+    TransactionFormUtils.onInit(
+      this,
+      this.childForm_2,
+      this.childTransaction_2,
+      this.childContactIdMap_2,
+      this.contactService
+    );
+    TransactionChildFormUtils.childOnInit(this, this.childForm_2, this.childTransaction_2);
+  }
+
+  override ngOnDestroy(): void {
+    super.ngOnDestroy();
+    Object.values(this.childContactIdMap_2).forEach((id$) => id$.complete());
+  }
+
+  override save(navigationEvent: NavigationEvent) {
+    // update all contacts with changes from form.
+    if (this.transaction && this.childTransaction && this.childTransaction_2) {
+      TransactionContactUtils.updateContactsWithForm(this.transaction, this.templateMap, this.form);
+      TransactionContactUtils.updateContactsWithForm(this.childTransaction, this.childTemplateMap, this.childForm);
+      TransactionContactUtils.updateContactsWithForm(
+        this.childTransaction_2,
+        this.childTemplateMap_2,
+        this.childForm_2
+      );
+    } else {
+      throw new Error('Fecfile: No transactions submitted for triple-entry transaction form.');
+    }
+
+    const payload: Transaction = TransactionFormUtils.getPayloadTransaction(
+      this.transaction,
+      this.form,
+      this.formProperties
+    );
+
+    payload.children = [
+      TransactionFormUtils.getPayloadTransaction(this.childTransaction, this.childForm, this.childFormProperties),
+      TransactionFormUtils.getPayloadTransaction(this.childTransaction_2, this.childForm_2, this.childFormProperties_2),
+    ];
+    payload.children[0].report_id = payload.report_id;
+    payload.children[1].report_id = payload.report_id;
+
+    if (payload.transaction_type_identifier) {
+      const responseFromApi = this.writeToApi(payload);
+      responseFromApi.subscribe((transaction) => {
+        navigationEvent.transaction = this.transactionType?.updateParentOnSave ? payload : transaction;
+        this.navigateTo(navigationEvent);
+      });
+    }
+  }
+
+  override handleNavigate(navigationEvent: NavigationEvent): void {
+    this.formSubmitted = true;
+
+    if (navigationEvent.action === NavigationAction.SAVE) {
+      if (
+        this.form.invalid ||
+        this.childForm.invalid ||
+        this.childForm_2.invalid ||
+        !this.transaction ||
+        !this.childTransaction ||
+        !this.childTransactionType_2
+      ) {
+        return;
+      }
+
+      let confirmation$ = this.confirmWithUser(this.transaction, this.form);
+      confirmation$ = concat(
+        confirmation$,
+        this.confirmWithUser(this.childTransaction, this.childForm, 'childDialog')
+      ).pipe(reduce((accumulator, confirmed) => accumulator && confirmed));
+      if (this.childTransaction_2) {
```

Comment:
you're already checking if childTransaction_2 exists at line 135

Reply:
Fixed